Assets/Scripts/EnemyController.cs

```diff
@@ -47,9 +52,13 @@ public class EnemyController : MonoBehaviour
     [SerializeField]
     private bool showEscapeRadius;
 
-    private Transform _target;
+    // component references
     private NavMeshAgent _agent;
     private WanderAI _wanderAI;
+    private AttackLogic _attackLogic;
```

Comment:
That's a good point, but AttackLogic needs IMovable anyway, so that it has a target which it can turn too and which it can check the position after completing the hit :)

I look at it like this: 
You click on the ground --> you move --> Controller
You click on an enemy --> you attack --> AttackLogic
If you have to move in order to attack the enemy that's part of the AttackLogic. Otherwise you'd have to implement something like "target out of range" when clicking on an enemy far away.

Reply:
Ahh I see, I was always thinking this way:
You click on something: ground or enemy or resource (Not in range) -->Move to it --> Movable
You click on something (In range) --> Trigger action (e.g. attack or collect) --> specific script/component
Both works I guess